### PR TITLE
DOC: Fix errors detected by sphinx-lint

### PR DESCRIPTION
**Pull request by matilda-me**
Found a few issues on the doc with [sphinx-lint](https://github.com/sphinx-contrib/sphinx-lint).

Fixed them.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Nice, thanks - could you add this to https://github.com/pandas-dev/pandas/blob/672798ec7b5565366be24aa4f57f6374f3a5e0fd/.pre-commit-config.yaml

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -525,7 +525,7 @@ Use :meth:`arrays.IntegerArray.to_numpy` with an explicit ``na_value`` instead.
 
    a.to_numpy(dtype="float", na_value=np.nan)
 
-**Reductions can return ``pd.NA``**
+**Reductions can return ``pd.NA``\ **
```
Comment (datapythonista) [inline doc/source/whatsnew/v1.0.0.rst:528]: I'm not sure if monospace texts can be bold to. Maybe the next is a simpler solution?

```suggestion
**Reductions can return** ``pd.NA``
```

I'm assuming the backslash space tells sphinx that there is nothing separating those two formatting options, tight?

**Review by datapythonista — APPROVED** [review]
Really nice clean up. I thought we were already validating this in the CI, but since it's not will be great to have it added to `precommit` as said. But I think this can be in a follow up PR.

Thanks a lot!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -70,6 +70,10 @@ repos:
       - id: rst-inline-touching-normal
         types: [text]  # overwrite types: [rst]
         types_or: [python, rst]
+-   repo: https://github.com/sphinx-contrib/sphinx-lint
+    rev: main
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:74]: can you put a tag (or a commit)? `main` isn't supported

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:74]: https://pre-commit.com/#using-the-latest-version-for-a-repository

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (matilda-me) [reply doc/source/whatsnew/v1.0.0.rst]: Yes backslash-space is "nothing" in RST.

Comment (matilda-me) [reply doc/source/whatsnew/v1.0.0.rst]: But I used your proposition, it's cleaner to read as plain text.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (matilda-me) [reply .pre-commit-config.yaml]: I used a tag instead.

**Review by MarcoGorelli — APPROVED** [review]
nice, looks good to me pending green

**Comment by matilda-me** [comment]
I don't understand the error :(

**Comment by MarcoGorelli** [comment]
looks unrelated, don't worry about it

**Comment by matilda-me** [comment]
Should I merge main to get it fixed (in other words, do I have something to do?)

**Comment by MarcoGorelli** [comment]
I re-ran and it passed - all good, thanks!

**Pull request merged**